File: 프로그래머스/2/12936. 줄 서는 방법/줄 서는 방법.py
from math import factorial


def solution(n, k):
    answer = []
    nums = list(range(1, n + 1))
    k -= 1
    
    while nums:
        div, k = divmod(k, factorial(len(nums) - 1))
        answer.append(nums.pop(div))
    
    return answer

File: 프로그래머스/2/12936. 줄 서는 방법/줄 서는 방법.java
import java.util.*;

class Solution {
    public int[] solution(int n, long k) {
        int[] answer = new int[n];
        List<Integer> nums = new ArrayList<>();
        
        for (int i = 1; i <= n; i++) {
            nums.add(i);
        }
        
        k--;
        
        for (int i = 0; i < n; i++) {
            long fact = factorial(n - 1 - i);
            int index = (int)(k / fact);
            k %= fact;
            answer[i] = nums.remove(index);
        }
        
        return answer;
    }

    private long factorial(int num) {
        long result = 1;
        for (int i = 2; i <= num; i++) {
            result *= i;
        }
        return result;
    }
}